sim-lib/src/lib.rs

```diff
@@ -918,8 +952,27 @@ async fn produce_events<N: DestinationGenerator + ?Sized, A: PaymentGenerator +
     sender: Sender<SimulationEvent>,
     listener: Listener,
 ) -> Result<(), SimulationError> {
+    let mut current_count = 0;
     loop {
-        let wait = node_generator.next_payment_wait();
+        if let Some(c) = node_generator.payment_count() {
+            if c == current_count {
+                log::info!(
+                    "Payment count has been met for {source}: {c} payments. Stopping the activity."
+                );
+                return Ok(());
+            }
+        }
+
+        let wait: Duration = if current_count == 0 {
+            log::debug!(
+                "Using a start delay. The first payment for {source} will be at {:?} seconds.",
```

Comment:
nit: this will log even if no delay is set:
` Using a start delay. The first payment for alice(020f3b...b44c96) will be at 0ns seconds.`

move `node_generator.payment_start()` up and log/use if !=0?

Reply:
Ah good catch. I just added a check to only log that if the start time != 0. Also moved the `Next payment for {source}...` message into the `else` block because it was kind of redundant at the start of the simulation. Also removed the word `seconds` from both of those log messages because the `Duration` class prints the units automatically, so it was also redundant.